client/src/Auth/Register/register.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+// https://www.flaticon.com/free-icon/circular-avatar_108331
+import avatar from './avatar.png';
+import './register.css';
+
+const Register = (props) => {
+    return (
+        <div className="container-fluid d-flex flex-column my-auto">
+                <div className="d-flex flex-column col-lg-5 col-md-3 col-sm-8 mx-auto justify-content-center register-wrapper">
+                    <div className="align-center pt-2 mb-3">
+                        <h2 className="register-text">Register</h2>
+                    </div>
+                    <img src={avatar} className="align-self-center my-3" width="120px" />
+                    <label>Username</label>
+                    <input type="text" placeholder="Username" className="form-control mb-2" value={props.inputs.username} onChange={(e) => {props.handleChange('username', e)} } />
+                    <label>Password</label>
+                    <input type="password" placeholder="Password" className="form-control mb-2" value={props.inputs.password} onChange={(e) => {props.handleChange('password', e)} }/>
+                    { props.inputs.password && 
+                        <div>
+                            <label>Confirm Password</label>
+                            <input type="password" placeholder="Confirm Password" className="form-control mb-2" value={props.inputs.confirmPassword} onChange={(e) => {props.handleChange('confirmPassword', e)} }/>
+                        </div>
+                    }
+                    <button disabled={(props.inputs.username.length < 2) && (props.inputs.password !== props.inputs.confirmPassword)} className="btn btn-outline-primary" onClick={props.handleSubmit}>Submit</button>
```

Comment:
That should be an `||`, not an `&&`. The button is almost never disabled.

Reply:
No, it is definitely intended to be an `&&` - for people registering we want both the username to be 3 or more characters and also we want them to have a password which they can make sure matches.  However I could add more prompts in order to inform the user what is necessary to register.

Reply:
You are testing on the opposite. Your code says: disable this button if both conditions are true:
- the username is less than 2 (!) chars
AND
- the config password doesn't match the password

What you actually want to do is disable the button when either:
- the username is less than 3 char
AND/OR
- the passwords don't match

Definitely an OR.

And yes, definitely tell the user what is wrong. When the passwords don't match but are the same length, the user can't spot it.

Reply:
My mistakes because we wanted to evaluate them both to `false` however I changed the way I did it to be more manageable hopefully.